File: AndroidLearnProject/app/src/main/learn/activity/FontSizeActivity.java
package com.android.learn.activity;

import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.os.Handler;
import android.view.KeyEvent;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.android.learn.R;
import com.android.learn.base.activity.BaseActivity;
import com.android.learn.base.activity.BaseMvpActivity;
import com.android.learn.base.event.RestartMainEvent;
import com.android.learn.base.mpresenter.BasePresenter;
import com.android.learn.base.utils.LanguageUtil;
import com.android.learn.base.utils.LogUtil;
import com.android.learn.base.utils.SPUtils;
import com.android.learn.base.utils.ScreenUtils;
import com.android.learn.view.fontsliderbar.FontSliderBar;

import org.greenrobot.eventbus.EventBus;

import butterknife.BindView;
import butterknife.OnClick;



public class FontSizeActivity extends BaseMvpActivity {
    @BindView(R.id.iv_back)
    ImageView iv_back;
    @BindView(R.id.title)
    TextView title;
    @BindView(R.id.fontSliderBar)
    FontSliderBar fontSliderBar;
    @BindView(R.id.tv_chatcontent1)
    TextView tvContent1;
    @BindView(R.id.tv_chatcontent)
    TextView tvContent2;
    @BindView(R.id.iv_userhead)
    ImageView ivUserhead;
    private float textsize1, textsize2, textsize3;
    private float textSizef;//缩放比例
    private boolean isClickable = true;
    int currentIndex;
    String TAG="FontSizeActivity";

    public static void startActivity(Context context) {
        Intent intent = new Intent(context, FontSizeActivity.class);
        context.startActivity(intent);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_fontsizes;
    }

    @Override
    protected void initData(Bundle bundle) {
        title.setText(getString(R.string.font_size));
        iv_back.setVisibility(View.VISIBLE);
        initData();
    }

    private void initData() {
        currentIndex = (Integer) SPUtils.getParam(this, "currentIndex", 1);
        textSizef = 1 + currentIndex * 0.1f;
        float size1 = tvContent1.getTextSize();
        float size2 = tvContent2.getTextSize();
        textsize1 = size1 / textSizef;
        textsize2 = size2 / textSizef;
        fontSliderBar.setTickCount(6).setTickHeight(ScreenUtils.dp2px(FontSizeActivity.this, 15)).setBarColor(Color.GRAY)
                .setTextColor(Color.BLACK).setTextPadding(ScreenUtils.dp2px(FontSizeActivity.this, 10)).setTextSize(ScreenUtils.dp2px(FontSizeActivity.this, 14))
                .setThumbRadius(ScreenUtils.dp2px(FontSizeActivity.this, 10)).setThumbColorNormal(Color.GRAY).setThumbColorPressed(Color.GRAY)
                .setOnSliderBarChangeListener(new FontSliderBar.OnSliderBarChangeListener() {
                    @Override
                    public void onIndexChanged(FontSliderBar rangeBar, int index) {
                        if (index > 5) {
                            return;
                        }
                        index = index - 1;
                        float textSizef = 1 + index * 0.1f;
                        setTextSize(textSizef);
                    }
                }).setThumbIndex(currentIndex).withAnimation(false).applay(this);

    }

    @OnClick({R.id.iv_back})
    public void click(View view) {

        switch (view.getId()) {
            case R.id.iv_back:
                if (fontSliderBar.getCurrentIndex() != currentIndex) {
                    if (isClickable) {
                        isClickable = false;
                        refresh();
                    }
                } else {
                    finish();
                }
                break;

        }
    }

    private void setTextSize(float textSize) {
        //改变当前页面的字体大小
        float size1 = textsize1 * textSize;
        float size2 = textsize2 * textSize;
        float size3 = textsize3 * textSize;
        tvContent1.setTextSize(ScreenUtils.px2sp(FontSizeActivity.this, size1));
        tvContent2.setTextSize(ScreenUtils.px2sp(FontSizeActivity.this, size2));
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            if (currentIndex != fontSliderBar.getCurrentIndex()) {
                if (isClickable) {
                    isClickable = false;
                    refresh();
                }
            } else {
                finish();
            }
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }

    private void refresh() {
        //存储标尺的下标
        SPUtils.setParam(this, "currentIndex", fontSliderBar.getCurrentIndex());
        //通知主页面重启
        EventBus.getDefault().post(new RestartMainEvent(this));

    }

    public void onDestroy(){
        super.onDestroy();
        LogUtil.d(TAG,TAG+"   onDestroy--------");
    }

    @Override
    public BasePresenter initPresenter() {
        return null;
    }

    @Override
    protected void loadData() {

    }
}

File: app/src/main/java/com/android/learn/activity/FontSizeActivity.kt
package com.android.learn.activity

import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import butterknife.OnClick
import com.android.learn.R
import com.android.base.activity.BaseActivity
import com.android.base.event.RestartMainEvent
import com.android.base.utils.LogUtil
import com.android.base.utils.SPUtils
import com.android.base.utils.ScreenUtils
import com.android.learn.view.fontsliderbar.FontSliderBar
import org.greenrobot.eventbus.EventBus


class FontSizeActivity : BaseActivity() {
    @BindView(R.id.iv_back)
    lateinit var iv_back: ImageView
    @BindView(R.id.title)
    lateinit var title: TextView
    @BindView(R.id.fontSliderBar)
    lateinit var fontSliderBar: FontSliderBar
    @BindView(R.id.tv_chatcontent1)
    lateinit var tvContent1: TextView
    @BindView(R.id.tv_chatcontent)
    lateinit var tvContent2: TextView
    @BindView(R.id.iv_userhead)
    lateinit var ivUserhead: ImageView
    private var textsize1: Float = 0.toFloat()
    private var textsize2: Float = 0.toFloat()
    private val textsize3: Float = 0.toFloat()
    private var textSizef: Float = 0.toFloat()//缩放比例
    private var isClickable = true
    internal var currentIndex: Int = 0
    internal var TAG = "FontSizeActivity"

    override val layoutId: Int
        get() = R.layout.activity_fontsizes

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun initData(bundle: Bundle?) {
        title.text = getString(R.string.font_size)
        iv_back.visibility = View.VISIBLE
        initData()
    }

    private fun initData() {
        currentIndex = SPUtils.getParam(this, "currentIndex", 1) as Int
        textSizef = 1 + currentIndex * 0.1f
        val size1 = tvContent1.textSize
        val size2 = tvContent2.textSize
        textsize1 = size1 / textSizef
        textsize2 = size2 / textSizef
        val listener = object : FontSliderBar.OnSliderBarChangeListener {
            override fun onIndexChanged(rangeBar: FontSliderBar, index: Int) {
                var index = index
                if (index > 5) {
                    return
                }
                index = index - 1
                val textSizef = 1 + index * 0.1f
                setTextSize(textSizef)
            }
        }
        fontSliderBar.setTickCount(6).setTickHeight(ScreenUtils.dp2px(this@FontSizeActivity, 15).toFloat()).setBarColor(Color.GRAY)
                .setTextColor(Color.BLACK).setTextPadding(ScreenUtils.dp2px(this@FontSizeActivity, 10)).setTextSize(
                ScreenUtils.dp2px(this@FontSizeActivity, 14))
                .setThumbRadius(ScreenUtils.dp2px(this@FontSizeActivity, 10).toFloat()).setThumbColorNormal(Color.GRAY).setThumbColorPressed(Color.GRAY)
                .setOnSliderBarChangeListener(listener).setThumbIndex(currentIndex).withAnimation(false).applay(this)

    }

    @OnClick(R.id.iv_back)
    fun click(view: View) {

        when (view.id) {
            R.id.iv_back -> if (fontSliderBar.currentIndex != currentIndex) {
                if (isClickable) {
                    isClickable = false
                    refresh()
                }
            } else {
                finish()
            }
        }
    }

    private fun setTextSize(textSize: Float) {
        //改变当前页面的字体大小
        val size1 = textsize1 * textSize
        val size2 = textsize2 * textSize
        val size3 = textsize3 * textSize
        tvContent1.textSize = ScreenUtils.px2sp(this@FontSizeActivity, size1).toFloat()
        tvContent2.textSize = ScreenUtils.px2sp(this@FontSizeActivity, size2).toFloat()
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            if (currentIndex != fontSliderBar.currentIndex) {
                if (isClickable) {
                    isClickable = false
                    refresh()
                }
            } else {
                finish()
            }
            return true
        }
        return super.onKeyDown(keyCode, event)
    }

    private fun refresh() {
        //存储标尺的下标
        SPUtils.setParam(this, "currentIndex", fontSliderBar.currentIndex)
        //通知主页面重启
        EventBus.getDefault().post(RestartMainEvent(this))

    }

    public override fun onDestroy() {
        super.onDestroy()
        LogUtil.d(TAG, "$TAG   onDestroy--------")
    }


    companion object {

        fun startActivity(context: Context) {
            val intent = Intent(context, FontSizeActivity::class.java)
            context.startActivity(intent)
        }
    }
}
